pennylane/ops/functions/eigvals.py

```diff
@@ -88,11 +91,15 @@ def circuit(theta):
     if isinstance(op, qml.Hamiltonian):
         warnings.warn(
             "For Hamiltonians, the eigenvalues will be computed numerically. "
-            "This may be computationally intensive for a large number of wires.",
+            "This may be computationally intensive for a large number of wires."
+            "Consider using a sparse representation of the Hamiltonian with qml.SparseHamiltonian.",
             UserWarning,
         )
         return qml.math.linalg.eigvalsh(qml.matrix(op))
 
+    if isinstance(op, qml.SparseHamiltonian):
+        return scipy.sparse.linalg.eigsh(op.sparse_matrix(), k=6, which="SA")[0]

```

Comment:
Maybe returning only 6 eigenvalues may give some problem in the future or somewhere in the current code 🤔 What do you think about this?

Reply:
I just saw that you have it set as "Possible Drawbacks", I wonder if it would be convenient to have a different function that returns the minimum eigenvalue K = 1, and leave this function as it was

Reply:
I agree, maybe even adding a comment in the doc string about this, so. users know that they will get at most the first 6 eigvals could suffice.

Reply:
The first thing I thought was that @Jaybsoni , put it in the doc. But maybe a method to calculate a certain expected value needs to get all the eigenvalues and it will give error.

Reply:
@josh146, yes.

Reply:
@KetpuntoG oh I mean that more so that if the users does `qml.eigvals(obs, k=N)`, then we simply intervene, convert to to a dense matrix, and just use `np.linalg` 🙂

Reply:
@KetpuntoG, when the requested `k` is larger than `N`, `np.linalg.eigvalsh` can be used.

> Maybe even setting k=N results in np.linalg.eigvalsh being called.